galleries/examples/statistics/histogram_normalization.py

```diff
@@ -0,0 +1,225 @@
+# data. The value attached to each bar is divided by the total number of data
+# points _and_ the width of the bin, and thus the values _integrate_ to one
+# when integrating across the full range of data.
+# e.g. (``density = counts / (sum(counts) * np.diff(bins))``),
+# and (``np.sum(density * np.diff(bins)) == 1``).
+#
+# This normalization is how `probability density functions
+# <https://en.wikipedia.org/wiki/Probability_density_function>`_ are defined in
+# statistics.  If :math:`X` is a random variable on :math:`x`, then :math:`f_X`
+# is is the probability density function if :math:`P[a<X<b] = \int_a^b f_X dx`.
+# If the units of x are Volts, then the units of :math:`f_X` are :math:`V^{-1}`
+# or probability per change in voltage.
+#
+# The usefulness of this normalization is a little more clear when we draw from
+# a known distribution and try to compare with theory.  So, choose 1000 points
+# from a `normal distribution
+# <https://en.wikipedia.org/wiki/Normal_distribution>`_, and also calculate the
+# known probability density function:
+
+xdata = rng.normal(size=1000)
+xpdf = np.arange(-4, 4, 0.1)
+pdf = 1 / (np.sqrt(2 * np.pi)) * np.exp(-xpdf**2 / 2)
+
+# %%
+# to make the point very obvious, consider bins that do not have the same
+# spacing.  By normalizing by density, we preserve the shape of the
+# distribution, whereas if we do not, then the wider bins have much higher
+# values than the thin bins:
+
+fig, ax = plt.subplot_mosaic([['False', 'True']], layout='constrained')
+dx = 0.1
+xbins = np.hstack([np.arange(-4, -1.25, 6*dx), np.arange(-1.25, 4, dx)])
+ax['False'].hist(xdata, bins=xbins, density=False, histtype='step')
+ax['False'].set_ylabel('Count per bin')
+ax['False'].set_xlabel('x bins (below -1.25 bins are wider)')
+
+ax['True'].hist(xdata, bins=xbins, density=True, histtype='step')
+ax['True'].plot(xpdf, pdf)
+ax['True'].set_ylabel('Probability (per dx)')
+ax['True'].set_xlabel('x bins (below -1.25 bins are wider)')
+
+# %%
+# Using *density* also makes it easier to compare histograms with different bin
+# widths. Note that in order to get the theoretical distribution, we must
+# multiply the distribution by the number of data points and the bin width
+
+fig, ax = plt.subplot_mosaic([['False', 'True']], layout='constrained')
+
+# expected PDF
+ax['True'].plot(xpdf, pdf, '--', label='PDF', color='k')
+
+for nn, dx in enumerate([0.1, 0.4, 1.2]):
+    xbins = np.arange(-4, 4, dx)
+    # expected histogram:
+    ax['False'].plot(xpdf, pdf*1000*dx, '--', color=f'C{nn}')
+    ax['False'].hist(xdata, bins=xbins, density=False, histtype='step')
+
+    ax['True'].hist(xdata, bins=xbins, density=True, histtype='step', label=dx)
+
+# Labels:
+ax['False'].set_xlabel('x bins')
+ax['False'].set_ylabel('Count per bin')
+ax['True'].set_ylabel('Probability (per dx)')
+ax['True'].set_xlabel('x bins')
+ax['True'].legend(fontsize='small')
+
+# %%
+
+# Sometimes people want to normalize so that the sum of counts is one.  This is
```

Comment:
```suggestion
# Sometimes people want to normalize so that the sum of counts is one, which is known as 
# the `probability mass function <https://en.wikipedia.org/wiki/Probability_mass_function>`_.  This is
```
(maybe retype it as I do not know if the line length is OK now...)

Reply:
OK, I changed to 

```
# Sometimes people want to normalize so that the sum of counts is one.  This is
# analogous to a `probability mass function
# <https://en.wikipedia.org/wiki/Probability_mass_function>` for a discrete
# variable where the sum of probabilities for all the values equals one. 
```

because I don't think the PMF is actually a valid thing for continuous variables.  Not to discourage anyone from normalizing this way, but I think this is really only is useful if the data is discrete, and the discrete values are always the same.

Reply:
Work fine with me.

That was actually my use case. Quantization errors.